app/resources/views/pages/dominion/advisors/magic.blade.php

```diff
@@ -38,7 +38,7 @@
                                     <td>{{ $spellInfo['description'] }}</td>
                                     <td class="text-center">{{ $spell->duration }}</td>
                                     <td class="text-center">
-                                        @if ($spell->cast_by_dominion_id !== null)
+                                        @if ($spell->cast_by_dominion_id !== null && ($spell->cast_by_dominion_id == $selectedDominion->id || $spellCalculator->isSpellActive($selectedDominion, 'surreal_perception')))
```

Comment:
If you don't have Surreal Perception active, someone casts a spell on you, and you cast SP afterwards, you're able to see in retrospect who cast that spell [on you].

Is this intended behavior? Or something to fix later? We can leave it like this for now regardless

Reply:
Yes, that's how I intended it. I think it makes sense thematically and makes the spell more useful (which is the primary goal of the GPC's changes to it).

Currently, you can always see who cast it (and more importantly, your enemies can too), which is not how it was in DC. GPC can discuss this again after a round to try it out. I think that will be standard going forward, to review all changes from the previous round and see how they worked out.

Reply:
Alright no worries. Just checking if it's not an oversight. If the GPC likes it we keep it